pkg/sysinfo/sysinfo.go

```diff
@@ -102,6 +117,128 @@ type cgroupPids struct {
 	PidsLimit bool
 }
 
+// ValidateHugetlb check whether hugetlb pagesize and limit legal
+func (c cgroupHugetlbInfo) ValidateHugetlb(pageSize string, limit uint64) (string, []string, error) {
+	var (
+		w   []string
+		err error
+	)
+	if pageSize != "" {
+		if err = isHugepageSizeValid(pageSize); err != nil {
+			return "", w, err
+		}
+	} else {
+		pageSize, err = c.GetDefaultHugepageSize()
+		if err != nil {
+			return "", w, fmt.Errorf("Failed to get system hugepage size")
```

Comment:
Do we actually need to know the default pagesize if no size is specified? reading the [documentation](https://github.com/torvalds/linux/blob/30955e71fc2bd1b966ce6531285ac59ae653819f/Documentation/vm/hugetlbpage.txt#L74-L76)

> Some platforms support multiple huge page sizes.  To allocate huge pages of a specific size, one must precede the huge pages boot command parameters with a huge page size selection parameter `"hugepagesz=<size>"`

i.e., if we want to use the default, do we have to specify anything?

Reply:
users do not have to specify, but we need to get default size to write `/sys/fs/cgroup/hugetlb/hugetlb.XXXX.limit_in_bytes` file